lib/client/keystore.go

```diff
@@ -305,6 +305,7 @@ func (fs *FSLocalKeyStore) GetKey(idx KeyIndex, opts ...CertOption) (*Key, error
 		Priv:     priv,
 		TLSCert:  tlsCert,
 		TrustedCA: []auth.TrustedCerts{{
+			ClusterName:     idx.ClusterName,
```

Comment:
I don't think this is entirely correct. `TrustedCerts.ClusterName` should correspond to the cluster from which the host CA certs were obtained (which should be the root cluster) but `idx.ClusterName` is often set to a leaf cluster's name.

Reply:
good catch!
added a loop to properly split `tlsCA` into individual `auth.TrusterCert` based on the cluster name from the CN.

Reply:
Removed this change for now, turns out it's not needed.